### ISSUE TITLE
CachedSession.cache_disabled() cannot be nested

### ISSUE THREAD

**Issue by jhermann** [reclosedev/requests-cache#107]
This might appear academic, but if you think of more complex call trees, it's easy to nest two context managers into each other. The current code does not support this, and enables the cache after exiting any nested context, violating the assumptions of code higher up in the call tree.

The fix is easy:
```py
        if self._is_cache_disabled:
            yield
        else:
            self._is_cache_disabled = True
            try:
                yield
            finally:
                self._is_cache_disabled = False
```

**Comment by gchamon** [comment]
Is this what is causing the error `AttributeError: 'CachedSession' object has no attribute '_is_cache_disabled'` when using generators?

I use generators to unwind continuation tokens from an API. It makes retrieving a large set from this API transparent, without which the whole code would have to deal with the tokens explicitly, causing a lot of code duplication.

**Comment by JWCook** [comment]
@jhermann Thanks for the suggestion! This has been fixed.

@gchamon I believe that is a different issue, likely fixed by #163. Try out the changes from the master branch. If that didn't fix it, please create a new issue with a full code example.
